boa_engine/src/builtins/promise/mod.rs

```diff
@@ -0,0 +1,731 @@
+//! This module implements the global `Promise` object.
+
+#![allow(dead_code, unused_results, unused_variables)]
+
+#[cfg(test)]
+mod tests;
+
+mod promise_job;
+
+use boa_gc::{Finalize, Gc, Trace};
+use boa_profiler::Profiler;
+use tap::{Conv, Pipe};
+
+use crate::{
+    builtins::BuiltIn,
+    context::intrinsics::StandardConstructors,
+    job::JobCallback,
+    object::{
+        internal_methods::get_prototype_from_constructor, ConstructorBuilder, FunctionBuilder,
+        JsObject, ObjectData,
+    },
+    property::Attribute,
+    value::JsValue,
+    Context, JsResult,
+};
+
+use self::promise_job::PromiseJob;
+
+use super::JsArgs;
+
+/// JavaScript `Array` built-in implementation.
+#[derive(Debug, Clone, Copy)]
+pub(crate) struct Array;
```

Comment:
Why is this needed here?

Reply:
It is not, thanks for pointing this out :)

Reply:
Added the comment to #2107